Mathlib/Data/Finsupp/Defs.lean

```diff
@@ -455,6 +455,13 @@ theorem unique_single_eq_iff [Unique α] {b' : M} : single a b = single a' b' 
   rw [unique_ext_iff, Unique.eq_default a, Unique.eq_default a', single_eq_same, single_eq_same]
 #align finsupp.unique_single_eq_iff Finsupp.unique_single_eq_iff
 
+lemma apply_single [AddCommMonoid N] [AddCommMonoid P]
+    (e : N →+ P) (a : α) (n : N) (b : α) :
```

Comment:
Can you use `MonoidHomClass` here instead?

Reply:
without any difficulty.